TestHelpers/struct_field_to_index_map_generator.py

```diff
@@ -0,0 +1,312 @@
+
+    original_name_split = original_name.split(" ")
+    if len(original_name_split) > 1 and original_name_split[1] != "":
+        original_name = original_name_split[1]
+    else:
+        original_name = original_name_split[0]
+
+    typedef_alias_name = ""
+    start_index = typedef_decl.find("referenced") + len("referenced") + 1
+    cur_char = typedef_decl[start_index]
+
+    while (cur_char != " "):
+        typedef_alias_name += cur_char
+
+        start_index += 1
+        cur_char = typedef_decl[start_index]
+
+    if original_name != typedef_alias_name:
+        structs_manager.get_struct(
+            original_name, False).add_typedef(typedef_alias_name)
+        logout(f"added typedef '{original_name}' '{typedef_alias_name}'")
+
+
+file_to_read = sys.argv[1]
+output_file = sys.argv[2]
+
+with open(file_to_read) as ast:
+    cur_struct = None
+    cur_top_decl = None  # FunctionDecl or RecordDecl (strings)
+    cur_mid_decl = None  # ParmVarDecl or FieldDecl (strings)
+    field_index = None
+
+    ast_lines = ast.readlines()
+    structs_manager = StructsManager()
+    struct_var_manager = StructVarManager()
+
+    for expr in ast_lines:
+        expr = expr.strip()
+
+        first_letter_index = expr.find(next(filter(str.isalpha, expr)))
+        start_decl_index = first_letter_index
+        decl = ""
+        cur_decl_char = expr[start_decl_index]
+
+        try:
+            while (cur_decl_char != " "):
+                decl += cur_decl_char
+                start_decl_index += 1
+                cur_decl_char = expr[start_decl_index]
+        except IndexError:
+            # might be a null stement (looks like "<<<NULL>>>" in AST)
+            field_index = None
+            continue
+
+        if "FunctionDecl" == decl:
+            cur_top_decl = expr
+            field_index = None
+
+        elif "RecordDecl" == decl:
+            cur_top_decl = expr
+            field_index = None
+            cur_struct = add_struct(cur_top_decl, structs_manager)
+
+        elif "TypedefDecl" == decl:
+            cur_top_decl = expr
+            field_index = None
+            add_typedef_alias(structs_manager, cur_top_decl)
+
+        elif "ParmVarDecl" == decl:
+            cur_mid_decl = expr
+            field_index = None
+
+        elif "FieldDecl" == decl:
+            cur_mid_decl = expr
+
+            if field_index is None:
+                field_index = 0
+            else:
+                field_index += 1
+
+            add_field(cur_mid_decl, field_index, cur_struct)
+
+        elif "AnnotateAttr" == decl:
+            if "|-" not in expr:
+                cur_mid_decl = None
+
+        else:
+            if "VarDecl" == decl and "struct" in expr:
+                add_struct_var(expr[first_letter_index:],
+                               struct_var_manager, structs_manager)
+
+            field_index = None
+
+    with open(output_file, "w") as output:
+        for i in range(len(struct_var_manager.struct_vars)):
```

Comment:
pylint suggests:
```
struct_field_to_index_map_generator.py:300:8: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
struct_field_to_index_map_generator.py:306:12: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
```

I agree.